### PR TITLE
Add methods array.front() and array.back()

### PR DESCRIPTION
**Pull request by daviditen**
These methods return the first or last element in a 1-D rectangular array.
It is a compiler error to call them on other array types.  It is a runtime
error to call them on an empty array when bounds checking is enabled.

Added tests for these errors and a working case.

### CODE REVIEW

**Comment by daviditen** [comment]
This is for issue #8620.

**Comment by daviditen** [comment]
Thanks!

File: modules/internal/ChapelArray.chpl [hunk 1]
```diff
@@ -2795,6 +2795,19 @@ module ChapelArray {
...
 
+    /* Return the last element in the array. The array must be a
+       rectangular 1-D array.
+     */
+    proc back() {
+      if !isRectangularArr(this) || this.rank != 1 then
+        compilerError("back() is only supported on 1D rectangular arrays");
+
+      if boundsChecking && isEmpty() then
+        halt("back called on an empty array");
```
Comment (BryantLam) [inline modules/internal/ChapelArray.chpl]: The `halt` on really should be `throw`.

Comment (daviditen) [reply modules/internal/ChapelArray.chpl]: PR #10191 just added halt wrappers for this case, so it should call one of them.  Eventually we may want it to be a `throw`, but not quite yet.

**Pull request merged**